src/scripting_api/error.js

```diff
@@ -0,0 +1,23 @@
+/* Copyright 202O Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class NotSupportedError extends Error {
```

Comment:
Are you *absolutely* sure that this works correctly, even in the *transpiled* builds?

Please note that we had to jump through some hoops, in order for things to work everywhere, for the existing PDF.js `Exception` types; see https://github.com/mozilla/pdf.js/blob/88f72d6b1c37dfc0dbb6715685bf8cb4ae516cd7/src/shared/util.js#L419-L432

Hence I'd suggest that you simply place a copy of the `BaseException` class in the utility file in this directory and use that one here instead.

Reply:
> The exception is only thrown inside the sandbox (so for now the Firefox one).

Sure, but eventually this is intended to support e.g. the Chromium extension as well, according to https://github.com/mozilla/pdf.js/pull/12432#issue-496212813, and there the code is transpiled by default for compatibility reason.

However, I suppose that you don't actually *need* to do anything here right now, and this can be changed down the road if that actually turns out to be necessary :-)